Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/prefetch/PrefetchingStrategy.java

```diff
@@ -0,0 +1,135 @@
+package nl.vu.cs.s2group.nappa.prefetch;
+
+import androidx.annotation.NonNull;
+
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+import nl.vu.cs.s2group.nappa.graph.ActivityNode;
+
+/**
+ * The <tt>PrefetchStrategy</tt> interface provides the API to access and initialize
+ * the implemented prefetching strategies. The strategy is selected upon initializing the
+ * library API via:
+ * <br/><br/>
+ *
+ * <code>PrefetchingLib.init(STRATEGY_ID)</code>
+ * <br/><br/>
+ *
+ * <p>The available values of the {@code STRATEGY_ID} parameter are in the following list:
+ *
+ * <ul>
+ *     <li> {@link #STRATEGY_MOST_VISITED_SUCCESSOR}</li>
+ *     <li> {@link #STRATEGY_2}</li>
+ *     <li> {@link #STRATEGY_GREEDY}</li>
+ *     <li> {@link #STRATEGY_PPM} (Prediction by Partial Match)</li>
+ *     <li> {@link #STRATEGY_PAGERANK}</li>
+ *     <li> {@link #STRATEGY_HITS} (Hyperlink-Induced Topic Search)</li>
+ *     <li> {@link #STRATEGY_SALSA} (Stochastic Approach for Link-Structure Analysis)</li>
+ *     <li> {@link #STRATEGY_GREEDY_WITH_PAGERANK_SCORES}</li>
+ *     <li> {@link #STRATEGY_PPM_WITH_HITS_SCORES}</li>
+ * </ul>
```

Comment:
@iivanoo 
All strategies are named, except strategy 2, which would require further analysis or the source code.
I think that it is more clear now what the strategy does. For more information, this list is linked to each constant, which are also linked to the strategy implementation class.
For each class, I tried to add a description of the strategy.
The most detailed description is in the Greedy strategy, as described in Christian thesis.